src/lib/idleTimeout.js

```diff
@@ -79,6 +79,18 @@ export const handleIdleTimeout = async (idleTimeoutConfig) => {
 		const { token, room: { _id: rid } = {} } = store.state;
 		logger.info('Closing chat on widget timeout');
 
+		const loggerPayload = {
+			room_id: rid,
+			category: 'Chat Session',
+			action: 'closed',
+			properties: {
+				close_method: 'timeout',
+			},
+			event_type: 'session',
+			timestamp: new Date(),
+		};
+		Livechat.sendLogsToSns(loggerPayload);
```

Comment:
@ear-dev Here `close_method` is `timeout`, but it does not say using app or widget. Should we add one more property for it?

Reply:
And.... are we distinguishing between escalated timeout vs. visitor abandonment timeout?

Reply:
@ear-dev I mean the session got timed out from widget timer or timer running in the RC App

Reply:
>  And.... are we distinguishing between escalated timeout vs. visitor abandonment timeout?

The log point for escalated timeout will be in the App

Reply:
Thanks @Shailesh351 

We only use the timer in the app, so I think this is fine because I do not suspect that we will ever want to use the timer in the widget. 

My only question is how we will distinguish between VA timeout and SF timeout?  I understand that the events come from different sources, but on the back end is there a way to distinguish?  Or do we need to add something?  i.e. I don't think we have any indication in our payload to distinguish if the event is from the widget, the RC server, or one of the RC.apps?

Reply:
We can directly add a log point for the timeout in the SF App for SF timeout. And for the VA timeout may be server  will have handle on it or in the SF app we receive close chat event and we can use it somehow for identifying the VA timeout